improver/nbhood/nbhood.py

```diff
@@ -69,7 +68,7 @@ class BaseNeighbourhoodProcessing(PostProcessingPlugin):
 
     def __init__(
         self,
-        neighbourhood_method: Union[CircularNeighbourhood, SquareNeighbourhood],
+        neighbourhood_method: Union[Neighbourhood],
```

Comment:
I don't think you need to pass a method in like this anymore. It is always this plugin that we are using now, just configured for either square or circular neighbourhooding. Passing in a string stating "square" or "circular" would be sufficient and remove some of the complexity around checking if this is a callable etc.